tools/code-analyzer/src/lib/scan-changes.ts

```diff
@@ -22,11 +22,15 @@ export const scanForChanges = async (
 	skipSchemaCheck: boolean,
 	source: string,
 	base: string,
-	outputStyle: string
+	outputStyle: string,
+	clonedPath?: string
 ) => {
-	Logger.startTask( `Making temporary clone of ${ source }...` );
-	const tmpRepoPath = await cloneRepo( source );
-	Logger.endTask();
+	let tmpRepoPath = clonedPath;
+	if ( ! clonedPath ) {
+		Logger.startTask( `Making temporary clone of ${ source }...` );
+		tmpRepoPath = typeof clonedPath !== 'undefined' ? clonedPath : await cloneRepo( source );
+		Logger.endTask();
+	}
```

Comment:
This is not type safe and TS complains. I think everything done here could be wrapped into the const declaration and it would be typesafe:

You don't need to worry about checking strictly for undefined.

```suggestion
	Logger.startTask( `Making temporary clone of ${ source }...` );

	const tmpRepoPath =
		typeof clonedPath !== 'undefined'
			? clonedPath
			: await cloneRepo( source );

	Logger.notice(
		`Temporary clone of ${ source } created at ${ tmpRepoPath }`
	);
```